### ISSUE TITLE
Debugging AWS Lambda with specific profile

### ISSUE THREAD

**Issue by efremsternbach** [aws/aws-toolkit-vscode#1184]
**Desktop (please complete the following information):**

-   OS: Windows 10 Pro
-   VS Code version: 1.47
-   AWS Toolkit extension version: 1.10

**Question**

I've been successfully using VSC to debug and deploy python lambda functions in AWS. Recently I needed to set up work in multiple accounts (Production and Development). When I deploy the lambda it identifies the correct profile and deploys to the correct account. However when I try to debug the lambda locally it always connects to my "default" account. 

I've been using the CodeLens feature in my debugging. I understand one can also setup a separate launch configuration but I don't understand why the debugger is not using the correct profile. I think this is a bug but it's possible there is some configuration that could fix this.  Would appreciate any help.

**Comment by awschristou** [comment]
Thanks for reaching out @efremsternbach -- local debugging does not currently have profile support, so it would have used the `default` profile.

We've been working on a new local debugging experience, which also supports alternate profiles. If you are interested in trying it out and sharing feedback, you can find it over at #1146 . It relies more on (launch.json based) debug configurations and less on CodeLenses, so you may need to reference the linked docs for details on how to use it.

**Comment by efremsternbach** [comment]
I was able to download the 1.11 snapshot and get the profile to work. However I have a few comments/suggestions on how to improve this.

- Needs some examples. I had to play around to find where to specify my profile.
- Right there are 2 mechanisms to connect to a profile, the 'AWS:Connect to AWS' command and the profile specification in the launch configuration. I think this will be confusing since I have to set my profile twice. Since the 'AWS:Deploy SAM Application' command is already picking up the profile I set at the Visual Studio Code level I think the debugger launch configuration should default to the profile from the same place.

Of course this is a very useful tool even if there are some inconveniences. Just FYI we are running separate accounts for production and development since we just can't risk an inadvertent modification of our production deployments. We are going to have to make extensive use of different profiles for development and testing. Anything that makes this more transparent and easier to use has a huge value for us.

Thanks

**Comment by efremsternbach** [comment]
While this "works" the interface  has a problem. When you click "Add Debug Configuration" it doesn't add a new configuration, it overwrites (erases) all the previously existing configurations.

**Comment by awschristou** [comment]
@efremsternbach Thanks for the heads up. Can you please confirm this as the repro case:

- have a form of populated launch.json file
- create or load a sam app
- click the "Add Debug Configuration" CodeLens above a function handler
- launch.json is overwritten with one new entry

cc @justinmk3

**Comment by efremsternbach** [comment]
Today the AWS Toolkit extension was overwritten with the official release of version 1.11.0. I am now back to square one since this release (as opposed to the snapshot in #1146) does not recognize profiles for credentials when debugging. I've checked the AWS Toolkit settings and the Aws:Profile setting is being set correctly.

I don't want to criticize the great work people are doing on this project but I have to admit I'm having a hard time understanding why this is so hard. For now I'm going to have to go back to manually changing my default profile when I have to switch between working on my production and development accounts. While there may be other priorities this would be a great productivity enhancer for me if it were implemented.

**Comment by justinmk3** [comment]
@efremsternbach I've updated https://github.com/aws/aws-toolkit-vscode/issues/1146 to point to a new prerelease, `1.12`. We haven't merged that branch with the recent 1.11 release, so the changes there aren't available yet. 

Because VSCode auto-updates to the stable release, the 1.11 prerelease that you had installed, was overwritten. Sorry for the confusion!

**Comment by cvharris** [comment]
@efremsternbach how did you configure the `launch.json` to use a specific profile? I tried placing `"profile": "derp"` in different parts of my debug configuration and it didn't work.

**Comment by justinmk3** [comment]
> how did you configure the `launch.json` to use a specific profile? I tried placing `"profile": "derp"` in different parts of my debug configuration and it didn't work.

@cvharris Try this:

```
            "aws": {
                "credentials": "profile:derp",
                "region": "us-east-1"
            }
```

Once we release this feature there will be full docs, but note also that there's a help doc when you mouseover the `credentials` field:

<img width="708" alt="Screen Shot 2020-07-28 at 7 43 38 AM" src="https://user-images.githubusercontent.com/55561878/88681271-3fb20000-d0a6-11ea-9a0f-79a28adab6fd.png">

All fields in the `aws-sam` launch config type are (or should be :) are available from auto-complete when editing `launch.json`, and VSCode by default will also show docs when cycling through the completion menu.

**Comment by justinmk3** [comment]
The feature described in https://github.com/aws/aws-toolkit-vscode/issues/1146 was released in Toolkit 1.12.

If the above reply didn't resolve your issue let us know. Thanks!
